src/core/stdc/errno.c

```diff
@@ -11,7 +11,14 @@
 
 #include <errno.h>
 
-
+#if __DragonFly__
+#define _errno errno
+#undef errno
+int * __error1(void) {
+    return (&_errno);
+}
+#define	errno		(* __error1())
```

Comment:
I had some real difficulty mapping the DragonFly errno.h/error() implementation to errno.d. Mostly because it already defined an __error() function. The only way i could make it to be behave was by recreating the macro, using a different __error1() function and map to that. (BTW: Trying to use the setErrno() did not work either).
If someone has a better/nicer solution, or would like me to move this 'implementation' into /src/core/sys/dragonflybsd for example. just let me know.

Reply:
Took a look at [the Dragonfly header](http://fxr.watson.org/fxr/source/sys/errno.h?v=DFBSD#L44): can't you just replace all this C renaming with `int * __error1(void) {return __error();}`, to turn the header-only `__error` into a compiled `__error1`, then reference it as you do from D?  What am I missing?

Reply:
@joakim-noah I had not considered wrapping it :-) Testing your solution now.